testing/factory/factory.go

```diff
@@ -687,15 +687,13 @@ func (factory *Factory) MakeUnitReturningPassword(c *gc.C, params *UnitParams) (
 		c.Assert(err, jc.ErrorIsNil)
 	}
 
-	if model.Type() == state.ModelTypeIAAS {
-		agentTools := version.Binary{
-			Number:  jujuversion.Current,
-			Arch:    arch.HostArch(),
-			Release: params.Application.CharmOrigin().Platform.OS,
-		}
-		err = unit.SetAgentVersion(agentTools)
-		c.Assert(err, jc.ErrorIsNil)
+	agentTools := version.Binary{
+		Number:  jujuversion.Current,
+		Arch:    arch.HostArch(),
+		Release: params.Application.CharmOrigin().Platform.OS,
 	}
+	err = unit.SetAgentVersion(agentTools)
+	c.Assert(err, jc.ErrorIsNil)
```

Comment:
Sidecar charms (deployed to CAAS models) to have agent versions, so we should not check the model type here